File: Estruturais/Proxy/Proxy01.js
// Objeto final
class ServirReal{
    executar(){
        console.log("Executando o serviçi real...")

    }
}

// Proxy 
class ProxyServico{
    constructor(){
        this.ServirReal = new ServirReal();

    }

    executar(){
        console.log("Antes da execução do serviço");
        this.ServirReal.executar();
        console.log("Depois da execução do serviço");
    }
}

// uso do proxy - cliente 
const proxy = new ProxyServico();
proxy.executar();

File: Estruturais/Proxy/Proxy01.java
// objeto real
class ServicoReal{
    public void executar(){
        System.out.println("executando o serviço real");
    }

    
}


//proxy

class ProxyServico{
    private ServicoReal servicoReal;


    public ProxyServico(){
        this.servicoReal = new ServicoReal();

    }

    public void executar(){
        System.out.println("Antes da execução do serviço");
        servicoReal.executar();
        System.out.println("Depois da Executar do Serviço");

    }

}

//uso proxy
class Proxy01 {
    public static void main(String[] args){
        ProxyServico proxy = new ProxyServico();
        proxy.executar();
    }
}